api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/RadiologyHandler.java

```diff
@@ -24,15 +64,314 @@ public String getServiceCategory() {
 
 	@Override
 	public DiagnosticReport getFHIRDiagnosticReportById(String id) {
+		return getFHIRDiagnosticReport(Context.getEncounterService().getEncounterByUuid(id));
+	}
+
+	/**
+	 * Getting the data from a third party server, storing metadata in OpenMRS and passing on the bundle to the requester.
+	 *
+	 * @param name Given Name of the Subject/Patient
+	 * @return A bundle of DiagnosticReport which is matching with Given Name
+	 */
+	@Override
+	public List<DiagnosticReport> getFHIRDiagnosticReportBySubjectName(String name) {
+		log.debug("In Radiology Handler : getFHIRDiagnosticReportBySubjectName : " + name);
+		ArrayList<DiagnosticReport> diagnosticReports = new ArrayList<DiagnosticReport>();
+
+		String serverBase = FHIRUtils.getDiagnosticReportRadiologyBaseServerURL();
+		ICriterion<ReferenceClientParam> diagnosticReportBySubject = DiagnosticReport.SUBJECT.hasChainedProperty(
+				Patient.GIVEN.matches().value(name));
+		ICriterion<TokenClientParam> diagnosticReportByService = DiagnosticReport.SERVICE.exactly().code("RAD");
+		Bundle bundle = FHIRRESTfulGenericClient.search(serverBase, DiagnosticReport.class, diagnosticReportBySubject,
+				diagnosticReportByService);
+		log.debug("Bundle size : " + bundle.size());
+
+		for (BundleEntry entry : bundle.getEntries()) {
+			IResource resource = entry.getResource();
+			log.debug("Resource Type : " + resource.getResourceName());
+			if (resource.getResourceName().equals("DiagnosticReport")) {
+				DiagnosticReport diagnosticReport = (DiagnosticReport) resource;
+				/*diagnosticReport = this.saveFHIRDiagnosticReport(diagnosticReport);
+				diagnosticReport = this.getFHIRDiagnosticReportById(diagnosticReport.getId().getIdPart());*/
+				diagnosticReports.add(diagnosticReport);
+			}
+		}
+
+		return diagnosticReports;
+	}
+
+	private DiagnosticReport getFHIRDiagnosticReport(Encounter omrsDiagnosticReport) {
+		log.debug("Laboratory Handler : GetFHIRDiagnosticReport");
```

Comment:
normally use if(log.isDebugEnable()) {
log.debug(msg)
}
tHis is to improve performance